From 1e15ad6f9ec1efad28e319a73fa6bd1da950120d Mon Sep 17 00:00:00 2001
From: "Hayataka@kitakar5525" <Hayataka@kitakar5525>
Date: Tue, 5 Mar 2019 20:37:23 +0900
Subject: [PATCH 19/32] ipts: revert suspend/resume mechanism

The suspend/resume mechanism was introduced on commit a3a3ed3
to fix touch inputs lost after suspend/hibernation.
Because I don't have this issue and I don't want to remove ipts
module before suspend like this systemd/system-sleep/sleep on
jakeday repository, I revert this commit.

References:
・updating 4.19 patches to fix touch and suspend · jakeday/linux-surface@a3a3ed3
https://github.com/jakeday/linux-surface/commit/a3a3ed3dc433ed3b654669bfb57089e32d7077eb
・systemd/system-sleep/sleep - linux-surface/sleep at 5.1.15-1 · jakeday/linux-surface
https://github.com/jakeday/linux-surface/blob/5.1.15-1/root/lib/systemd/system-sleep/sleep

・ Touchscreen (irrecoverably) lost after suspend/hibernate · Issue #270 · jakeday/linux-surface
https://github.com/jakeday/linux-surface/issues/270
・ [SP4] Hibernate doesn't turn off computer · Issue #320 · jakeday/linux-surface
https://github.com/jakeday/linux-surface/issues/320
・ [SP 5 (2017), kernel 4.18.20]: System freeze after resuming from hibernate · Issue #310 · jakeday/linux-surface
https://github.com/jakeday/linux-surface/issues/310
・ Can't wake up from suspend using lid or power button · Issue #381 · jakeday/linux-surface
https://github.com/jakeday/linux-surface/issues/381

---
 drivers/gpu/drm/i915/i915_drv.c   |  6 ------
 drivers/gpu/drm/i915/intel_ipts.c | 10 ----------
 drivers/gpu/drm/i915/intel_ipts.h |  2 --
 3 files changed, 18 deletions(-)

diff --git a/drivers/gpu/drm/i915/i915_drv.c b/drivers/gpu/drm/i915/i915_drv.c
index 2e56034f6823..d92ec4ddda9b 100644
--- a/drivers/gpu/drm/i915/i915_drv.c
+++ b/drivers/gpu/drm/i915/i915_drv.c
@@ -1906,9 +1906,6 @@ static int i915_drm_suspend(struct drm_device *dev)
 	struct pci_dev *pdev = dev_priv->drm.pdev;
 	pci_power_t opregion_target_state;
 
-	if (INTEL_GEN(dev_priv) >= 9 && i915_modparams.enable_guc && i915_modparams.enable_ipts)
-		intel_ipts_suspend(dev);
-
 	disable_rpm_wakeref_asserts(dev_priv);
 
 	/* We do a lot of poking in a lot of registers, make sure they work
@@ -2108,9 +2105,6 @@ static int i915_drm_resume(struct drm_device *dev)
 
 	enable_rpm_wakeref_asserts(dev_priv);
 
-	if (INTEL_GEN(dev_priv) >= 9 && i915_modparams.enable_guc && i915_modparams.enable_ipts)
-        intel_ipts_resume(dev);
-
 	return 0;
 }
 
diff --git a/drivers/gpu/drm/i915/intel_ipts.c b/drivers/gpu/drm/i915/intel_ipts.c
index b276a2f7839c..9899c91e4925 100644
--- a/drivers/gpu/drm/i915/intel_ipts.c
+++ b/drivers/gpu/drm/i915/intel_ipts.c
@@ -645,13 +645,3 @@ void intel_ipts_cleanup(struct drm_device *dev)
 		cancel_delayed_work(&intel_ipts.reacquire_db_work);
 	}
 }
-
-int intel_ipts_resume(struct drm_device *dev)
-{
-	return intel_ipts_init(dev);
-}
-
-void intel_ipts_suspend(struct drm_device *dev)
-{
-	intel_ipts_cleanup(dev);
-}
diff --git a/drivers/gpu/drm/i915/intel_ipts.h b/drivers/gpu/drm/i915/intel_ipts.h
index 45d7d1273adf..a6965d102417 100644
--- a/drivers/gpu/drm/i915/intel_ipts.h
+++ b/drivers/gpu/drm/i915/intel_ipts.h
@@ -28,8 +28,6 @@ struct drm_device;
 
 int intel_ipts_init(struct drm_device *dev);
 void intel_ipts_cleanup(struct drm_device *dev);
-int intel_ipts_resume(struct drm_device *dev);
-void intel_ipts_suspend(struct drm_device *dev);
 int intel_ipts_notify_backlight_status(bool backlight_on);
 int intel_ipts_notify_complete(void);
 
-- 
2.22.0

